File: src/Panorama.Web.Host/src/shared/service-proxies/scenography/dtos/ViewSceneDto.ts
import {IEntityDto} from "../../common/dtos/EntityDto";

export interface IViewSceneDto extends IEntityDto {
    correlationId: string;
    name: string;
    description: string | undefined;
    scenographyId: number;
    sceneData: string | undefined;
}

export class ViewSceneDto implements IViewSceneDto {
    id: number = undefined;
    correlationId: string = undefined;
    name: string = undefined;
    description: string | undefined = undefined;
    scenographyId: number = undefined;
    sceneData: string | undefined = undefined;
}

File: src/Panorama.Backing.Bus.Shared/Scenes/Dto/ViewSceneDto.cs
using Newtonsoft.Json;
using Panorama.Backing.Bus.Shared.Common.Dto;

namespace Panorama.Backing.Bus.Shared.Scenes.Dto;

[Serializable]
// TODO: do not expose db id's
public class ViewSceneDto : EntityDto<long>
{
    [JsonProperty("correlationId")]
    public string CorrelationId { get; set; }
    
    [JsonProperty("name")]
    public string Name { get; set; }
    
    [JsonProperty("description")]
    public string Description { get; set; }
    
    [JsonProperty("scenographyId")]
    // TODO: do not expose db id's
    public long ScenographyId { get; set; }
    
    [JsonProperty("sceneData")]
    public string SceneData { get; set; }
}